src/ledger/autofill.ts

```diff
@@ -0,0 +1,209 @@
+import BigNumber from "bignumber.js";
+import {
+  xAddressToClassicAddress,
+  isValidXAddress,
+} from "ripple-address-codec";
+
+import { Client } from "..";
+import { ValidationError } from "../common/errors";
+import { AccountInfoRequest, LedgerRequest } from "../models/methods";
+import { Transaction } from "../models/transactions";
+import { setTransactionFlagsToNumber } from "../models/utils";
+import { xrpToDrops } from "../utils";
+
+// 20 drops
+const LEDGER_OFFSET = 20;
+// 5 XRP
+const ACCOUNT_DELETE_FEE = 5000000;
+
+function scaleValue(value, multiplier, extra = 0): string {
+  return new BigNumber(value).times(multiplier).plus(extra).toString();
+}
+
+export interface ClassicAccountAndTag {
+  classicAccount: string;
+  tag: number | false | undefined;
+}
+
+/**
+ * Given an address (account), get the classic account and tag.
+ * If an `expectedTag` is provided:
+ * 1. If the `Account` is an X-address, validate that the tags match.
+ * 2. If the `Account` is a classic address, return `expectedTag` as the tag.
+ *
+ * @param Account - The address to parse.
+ * @param expectedTag - If provided, and the `Account` is an X-address,
+ *                    this method throws an error if `expectedTag`
+ *                    does not match the tag of the X-address.
+ * @returns The classic account and tag.
+ * @throws ValidationError when an address tag doesn't match the tag specificed in the transaction.
+ */
+function getClassicAccountAndTag(
+  Account: string,
+  expectedTag?: number
+): ClassicAccountAndTag {
+  if (isValidXAddress(Account)) {
+    const classic = xAddressToClassicAddress(Account);
+    if (expectedTag != null && classic.tag !== expectedTag) {
+      throw new ValidationError(
+        "address includes a tag that does not match the tag specified in the transaction"
+      );
+    }
+    return {
+      classicAccount: classic.classicAddress,
+      tag: classic.tag,
+    };
+  }
+  return {
+    classicAccount: Account,
+    tag: expectedTag,
+  };
+}
+
+/**
+ * Autofills fields in a transaction.
+ *
+ * @param client - A client.
+ * @param tx - A transaction to autofill fields.
+ * @param signersCount - A signers count used for multisign.
+ * @returns An autofilled transaction.
+ */
+async function autofill(
```

Comment:
The helper functions should all be above or below `autofill` - it's confusing when some are above and some are below